aggregator/src/aggregator.rs

```diff
@@ -307,17 +322,44 @@ impl<C: Clock> Aggregator<C> {
         aggregation_job_id: &AggregationJobId,
         req_bytes: &[u8],
         auth_token: Option<AuthenticationToken>,
+        taskprov_header: Option<&[u8]>,
```

Comment:
I'd prefer if we decoded the taskprov header as soon as possible, and then all these method signatures (this,  `handle_aggregate_continue`) would have the more expressive `taskprov: Option<&TaskConfig>`. It's also friendlier to tests, because constructing a fake `TaskConfig` value is easier than constructing the encoding thereof.

Many of these methods pass the request down as `&[u8]`, but we are forced to do that because often you can't parse the incoming request until you have the task, because sometimes you need to know the query type to parse the message. That isn't the case for the `TaskConfig`, so I'd like to parse it as soon as possible. Actually, I think it could be done in a trillium handler.